sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -75,12 +80,29 @@ case class UnaryMinus(child: Expression) extends UnaryExpression
       """})
     case _: CalendarIntervalType =>
       val iu = IntervalUtils.getClass.getCanonicalName.stripSuffix("$")
-      defineCodeGen(ctx, ev, c => s"$iu.negate($c)")
+      nullSafeCodeGen(ctx, ev, interval => s"""
+        try {
+          ${ev.value} = $iu.negate($interval);
```

Comment:
shall we have 2 methods `negate` and `safeNegate`? It's good to keep the generated code small.

Reply:
ok

Reply:
Seems a safeNegate need a patch for generated code, like
```java
s"""
  | if (${ev.value} = null) {
  |    (${ev.isNull} = true;
  |} 
   """
```, which does not make thing better